ui/core/individual_sim_ui.tsx

```diff
@@ -394,9 +397,10 @@ export abstract class IndividualSimUI<SpecType extends Spec> extends SimUI {
 			const savedSettings = window.localStorage.getItem(this.getSettingsStorageKey());
 			if (savedSettings != null) {
 				try {
-					const settings = IndividualSimSettings.fromJsonString(savedSettings, { ignoreUnknownFields: true });
+					const settingsProto = IndividualSimSettings.fromJsonString(savedSettings, { ignoreUnknownFields: true });
+					IndividualSimUI.updateProtoVersion(settingsProto);
 
-					this.fromProto(initEventID, settings);
+					this.fromProto(initEventID, settingsProto);
```

Comment:
The `updateProtoVersion()` call should go inside `fromProto()`, so that migrations will be robustly called whenever a proto is loaded, not just within the saved settings code.